### ISSUE TITLE
onPositionChanged callback looped & content jumping on first open

### ISSUE THREAD

**Issue by unixvb** [ammarahm-ed/react-native-actions-sheet#73]
The component is rendered straight under `SafeAreaProvider`.
```
<ActionSheet
      ref={ref}
      onPositionChanged={() => console.log('onPositionChanged')}
      onClose={() => console.log('onClose')}
      onOpen={() => console.log('onOpen')}>
      <Text>Test</Text>
      <Text>Test</Text>
      <Text>Test</Text>
    </ActionSheet>
```

![image](https://user-images.githubusercontent.com/10281464/116771563-33077180-aa55-11eb-92ec-637db90881d8.png)

This bug causes content jumping on the first `open`.
It's reproduced in ~80% of first renders.

Version used:
`"react-native-actions-sheet": "0.4.7",`

**Comment by unixvb** [comment]
There are 2 cases:
 1. `gestureEnabled={true}` - `onPositionChanged` called from `_onScroll` (much easier to reproduce).  
 2. `gestureEnabled={false}` - `onPositionChanged` called from `_showModal` && `_onScroll` (reproducing rarely).

**Comment by unixvb** [comment]
I figured out that first render content jumping and `onPositionChanged` are 2 different issues. 

`AnimatedSafeAreaView` `onLayout` is called due to some "magic" `height` changes on the first render. 
`containerStyle={{ height }}` do fix that.

Content container is jumping on the first render due to `transformValue` initial value
```this.transformValue = new Animated.Value(0);  // Container is opened```
on `_openAnimation`
```
this.transformValue.setValue(scrollOffset);   // Container is closed
Animated.parallel([
        Animated.spring(this.transformValue, {
          toValue: 0,                                               // Container is opening
          bounciness: bounceOnOpen ? bounciness : 1,
          speed: openAnimationSpeed,
          useNativeDriver: true,
        }),
        Animated.timing(this.opacityValue, {
          toValue: 1,
          duration: 150,
          useNativeDriver: true,
        }),
      ]).start();
```

**Comment by ammarahm-ed** [comment]
@unixvb try the example code because it works normally. Compare.

**Comment by efstathiosntonas** [comment]
@ammarahm-ed i've got the same issue and i've reverted to 0.4.2 to "fix" it. Strange this is that is not happening all the times, sometimes the sheet opens and starts jumping like crazy and some other times it opens just fine. I haven't changed anything when going from 04.2 to 0.4.7, it just started happening.

**Comment by ammarahm-ed** [comment]
> @ammarahm-ed i've got the same issue and i've reverted to 0.4.2 to "fix" it. Strange this is that is not happening all the times, sometimes the sheet opens and starts jumping like crazy and some other times it opens just fine. I haven't changed anything when going from 04.2 to 0.4.7, it just started happening.
> 
> demo video (0.4.7) (not "jumping" too much though, i've seen worst jumping without even "touching" it):
> 
> https://user-images.githubusercontent.com/717975/116999876-edfb6d80-ace8-11eb-8f9a-876531583083.mov
> 
> 

Is it happening only on iOS or android too?

**Comment by efstathiosntonas** [comment]
Haven’t tested it on Android, will test and provide feedback asap

**Comment by ammarahm-ed** [comment]
> > @ammarahm-ed i've got the same issue and i've reverted to 0.4.2 to "fix" it. Strange this is that is not happening all the times, sometimes the sheet opens and starts jumping like crazy and some other times it opens just fine. I haven't changed anything when going from 04.2 to 0.4.7, it just started happening.
> > 
> > demo video (0.4.7) (not "jumping" too much though, i've seen worst jumping without even "touching" it):
> > 
> > https://user-images.githubusercontent.com/717975/116999876-edfb6d80-ace8-11eb-8f9a-876531583083.mov
> > 
> > 
> 
> Is it happening only on iOS or android too? On Android i haven't seen it happen.

**Comment by ammarahm-ed** [comment]
> Haven’t tested it on Android, will test and provide feedback asap

It could also happen if the component inside actionsheet or the ActionSheet is being constantly rerendered. Also check that. The only reason this can happen is that when user scrolls, state is updating and onLayout is getting called over and over.

**Comment by efstathiosntonas** [comment]
I haven’t changed anything in these components and with 0.4.2 it works perfectly. I think the issue started when the height calculation was changed in this commit: https://github.com/ammarahm-ed/react-native-actions-sheet/commit/7424cb91e9bf672f7a2bafd22065b85cfa686969

**Comment by ammarahm-ed** [comment]
@efstathiosntonas Yes but it is only calculated once when the ActionSheet opens. I will investigate this more and see what might be the cause. I changed the calculation because it is more accurate when calculating the SafeAreaView top padding.

**Comment by efstathiosntonas** [comment]
@ammarahm-ed tried it on Android, no jumping issues so yes, it happens only on iOS

**Comment by ammarahm-ed** [comment]
> @ammarahm-ed tried it on Android, no jumping issues so yes, it happens only on iOS

Okay, can you post an example of your implementation that includes the props you have set on ActionSheet and the style of top most component inside it

**Comment by KendallWeihe** [comment]
This is happening to me as well. On iOS. Occurs intermittently. When it occurs, action sheet is "jumpy" (per previous descriptions) but also locks up my iOS Simulator. My CPU usage spikes to 80% and hovers there until I quit the Simulator.

I tried pegging to `0.4.2` but issue still seems to be occurring.

Implementation is simple: 

```typescript
<ActionSheet ref={actionSheetRef} gestureEnabled={true} onPositionChanged={() => {}}>
  <Things />
</ActionSheet>
```

**Comment by ammarahm-ed** [comment]
> This is happening to me as well. On iOS. Occurs intermittently. When it occurs, action sheet is "jumpy" (per previous descriptions) but also locks up my iOS Simulator. My CPU usage spikes to 80% and hovers there until I quit the Simulator.
> 
> I tried pegging to `0.4.2` but issue still seems to be occurring.
> 
> Implementation is simple: 
> 
> ```typescript
> <ActionSheet ref={actionSheetRef} gestureEnabled={true} onPositionChanged={() => {}}>
>   <Things />
> </ActionSheet>
> ```
> 
> **UPDATE** perhaps I was wrong, `0.4.2` does _seem_ to be working now. Deleted all cache, node modules etc & rebuilt. Seems to be working now.

Hey, so I have noticed that if you do something like this:
```jsx
<ActionSheet>
<View/>
<View/>
<View/>
</ActionSheet>
```
It will happen consistently. However if it's all wrapped in a View:
```jsx
<ActionSheet>
<View>
     <View/>
     <View/>
     <View/>
</View>
</ActionSheet>
```
It doesn't happen. Maybe I am wrong but let me know.

**Comment by ammarahm-ed** [comment]
> This is happening to me as well. On iOS. Occurs intermittently. When it occurs, action sheet is "jumpy" (per previous descriptions) but also locks up my iOS Simulator. My CPU usage spikes to 80% and hovers there until I quit the Simulator.
> 
> I tried pegging to `0.4.2` but issue still seems to be occurring.
> 
> Implementation is simple: 
> 
> ```typescript
> <ActionSheet ref={actionSheetRef} gestureEnabled={true} onPositionChanged={() => {}}>
>   <Things />
> </ActionSheet>
> ```
> 
> **UPDATE** perhaps I was wrong, `0.4.2` does _seem_ to be working now. Deleted all cache, node modules etc & rebuilt. Seems to be working now.

I have noticed that this doesn't happen if <Things/> are wrapped in a <View/>.

```jsx
​<​ActionSheet​ ​ref​=​{​actionSheetRef​}​ ​gestureEnabled​=​{​true​}​ ​onPositionChanged​=​{​(​)​ ​=>​ ​{​}​}​>​
 <View>
    ​<​Things​ ​/​>​
    <Things/>
</View>
​<​/​ActionSheet​>
```

**Comment by ammarahm-ed** [comment]
Hey @efstathiosntonas @KendallWeihe Can you test using master branch? I have fixed the problem I think.

``` bash
yarn add https://github.com/ammarahm-ed/react-native-actions-sheet#master
```
Deleted all cache, node modules etc & rebuilt

**Comment by KendallWeihe** [comment]
@ammarahm-ed Latest on master is working as intended. Can you ping this thread when you publish a package patch?

**Comment by ammarahm-ed** [comment]
@KendallWeihe I will close the issue when I release tomorrow so you will know.

**Comment by efstathiosntonas** [comment]
Thanks @ammarahm-ed, unfortunately I will be able to test in about 12hrs. Github is acting up and I just got the email about the issue updates.

**Comment by ammarahm-ed** [comment]
@efstathiosntonas No worries, I will wait until I get your input on it so I can publish the new version

**Comment by efstathiosntonas** [comment]
@ammarahm-ed just tested on iOS, looks good to me! Thanks for the fix!

**Comment by ammarahm-ed** [comment]
okay! Thanks for testing.

**Comment by ammarahm-ed** [comment]
Released `v0.4.8`

**Comment by calinpastean** [comment]
https://user-images.githubusercontent.com/55287082/119322970-df202f00-bc86-11eb-95d1-633546dc7688.mp4

Got a similar issue in 0.4.8 android only.

```    
<ActionSheet
      statusBarTranslucent
      containerStyle={styles.container}
      ref={actionSheetRef}
    >
      <BlurView
        style={styles.blurViewContainer}
        blurType="dark"
        blurAmount={10}
        reducedTransparencyFallbackColor="white"
      >
        {children}
      </BlurView>
    </ActionSheet>
```

and the styling (doesn't matter, it does this without any containerStyle prop:

```  
container: {
    backgroundColor: colors.transparent,
},
```
